x-pack/solutions/observability/plugins/infra/public/components/asset_details/tabs/dashboards/actions/actions.test.tsx

```diff
@@ -114,34 +115,61 @@ describe('Custom Dashboards Actions', () => {
     expect(screen.getByTestId('infraLinkDashboardMenu')).toBeDisabled();
     expect(screen.getByTestId('infraLinkDashboardMenu')).toHaveTextContent('Link new dashboard');
   });
-  it('should render the unlink dashboard action when the user can unlink a dashboard', () => {
-    jest.spyOn(hooks, 'useSavedObjectUserPermissions').mockImplementation(() => ({
-      canSave: true,
-      canDelete: true,
-    }));
-    render(
-      <UnlinkDashboard
-        onRefresh={() => {}}
-        assetType="host"
-        currentDashboard={TEST_CURRENT_DASHBOARD}
-      />
-    );
-    expect(screen.getByTestId('infraUnLinkDashboardMenu')).not.toBeDisabled();
-    expect(screen.getByTestId('infraUnLinkDashboardMenu')).toHaveTextContent('Unlink dashboard');
-  });
-  it('should render the unlink dashboard action when the user cannot unlink a dashboard', () => {
-    jest.spyOn(hooks, 'useSavedObjectUserPermissions').mockImplementation(() => ({
-      canSave: true,
-      canDelete: false,
-    }));
-    render(
-      <UnlinkDashboard
-        onRefresh={() => {}}
-        assetType="host"
-        currentDashboard={TEST_CURRENT_DASHBOARD}
-      />
-    );
-    expect(screen.getByTestId('infraUnLinkDashboardMenu')).toBeDisabled();
-    expect(screen.getByTestId('infraUnLinkDashboardMenu')).toHaveTextContent('Unlink dashboard');
+
+  describe('UnlinkDashboard', () => {
+    const fetchCustomDashboardsSpy = jest.spyOn(fetchCustomDashboards, 'useFetchCustomDashboards');
+
+    beforeEach(() => {
+      // provide mock for invocation to fetch custom dashboards
+      fetchCustomDashboardsSpy.mockReturnValue({
+        dashboards: [
+          {
+            id: 'test-so-id',
+            dashboardSavedObjectId: 'test-dashboard-id',
+            assetType: 'host',
+            dashboardFilterAssetIdEnabled: true,
+          },
+        ],
+        loading: false,
+        error: null,
+        // @ts-expect-error we provide a mock function as we don't need to test the actual implementation
+        refetch: jest.fn(),
+      });
+    });
+
+    afterEach(() => {
+      fetchCustomDashboardsSpy.mockClear();
+    });
```

Comment:
Added this in to fix the issue [here](https://buildkite.com/elastic/kibana-pull-request/builds/266993#0194655f-1466-4ee3-80ed-54e398b09492), the mocked hook here requires another hook that would have required a mock, hence the need to mock this.